### PR TITLE
feat: add donation voting merkle distribution strategy

### PR DESCRIPTION
**Pull request by ilpepepig**

### CODE REVIEW

File: contracts/strategies/DonationVotingOffchain.sol [hunk 1]
```diff
@@ -94,6 +94,8 @@ contract DonationVotingOffchain is CoreBaseStrategy, RecipientsExtension {
     uint64 public withdrawalCooldown;
     /// @notice amount to be distributed. `totalPayoutAmount` get reduced with each distribution.
     uint256 public totalPayoutAmount;
+    /// @notice If true, allocations are directly sent to recipients. Otherwise, they they must be claimed later.
+    bool public directTransfer;
```
Comment (0xOneTony) [inline contracts/strategies/DonationVotingOffchain.sol:98]: This should be immutable, just to be clear that it should not be changed

File: contracts/strategies/DonationVotingOffchain.sol [hunk 2]
```diff
@@ -262,23 +267,22 @@ contract DonationVotingOffchain is CoreBaseStrategy, RecipientsExtension {
...
-            amountAllocated[_recipients[i]][tokens[i]] += _amounts[i];
+            if (!directTransfer) amountAllocated[__recipients[i]][tokens[i]] += _amounts[i];
 
+            address recipientAddress = directTransfer ? _recipients[__recipients[i]].recipientAddress : address(this);
             if (tokens[i] == NATIVE) {
                 totalNativeAmount += _amounts[i];
+                if (directTransfer) SafeTransferLib.safeTransferETH(recipientAddress, _amounts[i]);
             } else {
-                SafeTransferLib.safeTransferFrom(tokens[i], _sender, address(this), _amounts[i]);
+                SafeTransferLib.safeTransferFrom(tokens[i], _sender, recipientAddress, _amounts[i]);
```
Comment (0xOneTony) [inline contracts/strategies/DonationVotingOffchain.sol:282]: Looks good! I would add a few comments here though

File: contracts/strategies/DonationVotingOffchain.sol [hunk 3]
```diff
@@ -197,12 +201,16 @@ contract DonationVotingOffchain is CoreBaseStrategy, RecipientsExtension {
         _updatePoolTimestamps(_registrationStartTime, _registrationEndTime);
     }
 
-    /// @notice Claim allocated tokens
-    /// @param _claims Claims to be claimed
-    function claimAllocation(Claim[] calldata _claims) external onlyAfterAllocation {
+    /// @param _data The data to be decoded
```
Comment (0xOneTony) [inline contracts/strategies/DonationVotingOffchain.sol:204]: missing notice, and would be great to add dev comments on this and all other affected functions on how directTransfer is being used, e.g. this function should not be called if directTransfer is true ...

File: contracts/core/libraries/Transfer.sol [hunk 4]
```diff
@@ -5,6 +5,11 @@ pragma solidity 0.8.19;
 import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
 // Internal Libraries
 import "./Native.sol";
+// Interfaces
+import {ISignatureTransfer} from "permit2/ISignatureTransfer.sol";
+import {IERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Permit.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IDAI} from "../../core/interfaces/IDAI.sol";
```
Comment (0xAustrian) [inline contracts/core/libraries/Transfer.sol:12]: Using `contracts/core/interfaces/IDAI.sol` would be better imho

File: contracts/strategies/DonationVotingMerkleDistribution.sol [hunk 5]
```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+// External Libraries
+import {MerkleProof} from "openzeppelin-contracts/contracts/utils/cryptography/MerkleProof.sol";
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+// Interfaces
+import {IAllo} from "../core/interfaces/IAllo.sol";
```
Comment (0xAustrian) [inline contracts/strategies/DonationVotingMerkleDistribution.sol:8]: As I said above using the remapping seems better.

File: contracts/strategies/DonationVotingMerkleDistribution.sol [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    /// @notice Flag to indicate whether the distribution has started or not.
+    bool public distributionStarted;
+
+    /// @notice The merkle root of the distribution will be set by the pool manager.
+    bytes32 public merkleRoot;
+
+    /// @notice This is a packed array of booleans to keep track of claims distributed.
+    /// @dev _distributedBitMap[0] is the first row of the bitmap and allows to store 256 bits to describe
+    /// the status of 256 claims
+    mapping(uint256 => uint256) private _distributedBitMap;
```
Comment (0xAustrian) [inline contracts/strategies/DonationVotingMerkleDistribution.sol:83]: If we want to use smock, `internal` is better than `private`.

File: test/foundry/integration/DonationVotingMerkleDistribution.t.sol [hunk 7]
```diff
@@ -0,0 +1,710 @@
...
+    DonationVotingMerkleDistribution,
+    DonationVotingOffchain
+} from "contracts/strategies/DonationVotingMerkleDistribution.sol";
+import {Errors} from "contracts/core/libraries/Errors.sol";
+import {IRecipientsExtension} from "contracts/extensions/interfaces/IRecipientsExtension.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IntegrationBase} from "./IntegrationBase.sol";
+
+contract IntegrationDonationVotingMerkleDistributionBase is IntegrationBase {
+    address internal constant NATIVE = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
```
Comment (0xAustrian) [inline test/foundry/integration/DonationVotingMerkleDistribution.t.sol:16]: You can add this to `IntegrationBase.sol`

**Review by 0xOneTony — APPROVED** [review]
:fire

**Review by 0xAustrian — APPROVED** [review]

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**